src/Options.php

```diff
@@ -729,6 +771,8 @@ private function configureOptions(OptionsResolver $resolver): void
             'capture_silenced_errors' => false,
             'max_request_body_size' => 'medium',
             'class_serializers' => [],
+            'http_connect_timeout' => HttpClientFactory::DEFAULT_HTTP_CONNECT_TIMEOUT,
+            'http_timeout' => HttpClientFactory::DEFAULT_HTTP_TIMEOUT,
```

Comment:
We can't put the defaults here, the default should be null. That's because otherwise you can't handle the situation where you don't have any concrete client available, but you have the option set, like with the HTTP proxy option.

Reply:
You'll need coalescing, yes. Because `null` is a valid value that indicates that the user hasn't provided any value.
This is needed due to the fact that you need to implement a check like this one: https://github.com/getsentry/sentry-php/blob/32e5415803ff0349ccb5e5b5e77b016320762786/src/HttpClient/HttpClientFactory.php#L138

Reply:
While the two options has been made `nullable`, the if above has not been changed to take this fact into account

Reply:
@Jean85 I think we should re-evaluate the fact that these two new options are `nullable`. It doesn't really make sense, as the timeout is a concept that generally speaking applies to any transport, be it HTTP or whatever else, and so they should always be set to a sensible value by default. Moreover, in retrospective, I'm starting to believe that there isn't a real benefit in throwing an error for cases like this one, and we may just silently ignore the error or document the behaviour rather than breaking the application at runtime. WDYT?

Reply:
This has been concluded to not have the options be nullable or throw exception.

> https://github.com/getsentry/sentry-php/pull/1282#issuecomment-1123322201